metpy/calc/tools.py

```diff
@@ -283,3 +283,42 @@ def reduce_point_density(points, radius, priority=None):
             keep[ind] = True
 
     return keep
+
+
+@exporter.export
+def log_interp(x, xp, fp, **kwargs):
+    r"""Interpolates data with logarithmic x-scale.
+
+    Interpolation on a logarithmic x-scale for interpolation values in pressure coordintates.
+
+    Parameters
+    ----------
+    x : array-like
+        The x-coordinates of the interpolated values.
+
+    xp : array-like
+        The x-coordinates of the data points.
+
+    fp : array-like
+        The y-coordinates of the data points, same length as cp.
+
+    Returns
+    -------
+    array-like
+        The interpolated values, same shape as x.
+
```

Comment:
One feedback we've gotten is that people would like more simple examples. Can you add a doctest-style example of usage here?